src/promptflow/promptflow/_sdk/_service/apis/collector.py

```diff
@@ -181,3 +187,56 @@ def _try_write_trace_to_cosmosdb(
         stack_trace = traceback.format_exc()
         logger.error(f"Failed to write trace to cosmosdb: {e}, stack trace is {stack_trace}")
         return
+
+
+def _try_get_blob_container_client(
```

Comment:
What's the latency for getting blob container client?
Do we need to move it in worker thread?

Reply:
Nice catch. Get default datastore may take 2~5s. We may have a cache for blob client.